backend/src/services/super-admin/super-admin-service.ts

```diff
@@ -70,10 +76,33 @@ export const superAdminServiceFactory = ({
     return newCfg;
   };
 
-  const updateServerCfg = async (data: TSuperAdminUpdate) => {
-    const updatedServerCfg = await serverCfgDAL.updateById(ADMIN_CONFIG_DB_UUID, data);
-    await keyStore.setItemWithExpiry(ADMIN_CONFIG_KEY, ADMIN_CONFIG_KEY_EXP, JSON.stringify(updatedServerCfg));
-    return updatedServerCfg;
+  const updateServerCfg = async (data: TUpdateServerCfgDTO) => {
+    let organization: TOrganizations | undefined;
+    if (data.defaultOrgSlug) {
+      organization = await orgDAL.findOne({
+        slug: data.defaultOrgSlug
+      });
+
+      if (!organization) {
+        throw new BadRequestError({
+          name: "Update server config",
+          message: "Failed to find default organization"
+        });
+      }
+    }
+
+    const updatedServerCfg = await serverCfgDAL.updateById(ADMIN_CONFIG_DB_UUID, {
+      ...omit(data, ["defaultOrgSlug"]),
+      defaultOrgId: organization?.id || null
```

Comment:
The issue is on patch if user didn't provide org id this would become null right. I think if its undefined then it must not do anything.

Reply:
This is actually alright. When the user hits the "Save" button in the UI and they already have a default org set, it'll be included in the request.

It will only be set to null if the user sets the field to an empty value in the UI @akhilmhdh